odbc/src/api/bind_col_tests.rs

```diff
@@ -466,4 +403,832 @@ mod unit {
+            let bound_buffer = s
+                .bound_cols
+                .read()
+                .unwrap()
+                .as_ref()
+                .unwrap()
+                .get(&1)
+                .unwrap()
+                .target_buffer;
+            assert_eq!(10, *(bound_buffer as *mut i32));
+            assert_eq!(20, *((bound_buffer as ULen + 4) as *mut i32));
+            assert_eq!(30, *((bound_buffer as ULen + 8) as *mut i32));
+            assert_eq!(40, *((bound_buffer as ULen + 12) as *mut i32));
+
+            // assert that the indicator has the correct values for column 1.
+            let indicator = s
+                .bound_cols
+                .read()
+                .unwrap()
+                .as_ref()
+                .unwrap()
+                .get(&1)
+                .unwrap()
+                .length_or_indicator;
+            assert_eq!(4, *indicator);
+            assert_eq!(4, *((indicator as ULen + 8) as *mut Len));
+            assert_eq!(4, *((indicator as ULen + 16) as *mut Len));
+            assert_eq!(4, *((indicator as ULen + 24) as *mut Len));
+
+            // assert that the first 4 values from the result set were put in the bound buffer array correctly for column 2.
+            let bound_buffer = s
+                .bound_cols
+                .read()
+                .unwrap()
+                .as_ref()
+                .unwrap()
+                .get(&2)
+                .unwrap()
+                .target_buffer;
+
+            // input_text_to_string_w requires a `usize` value; however, I need a negative value to test that the null termination character was added,
+            // So I input usize::MAX because it casts to -1 in the function.
+            assert_eq!(
+                "aaaa",
+                input_text_to_string_w(bound_buffer as *const WideChar, usize::MAX)
+            );
+            assert_eq!(
+                "bbbb",
+                input_text_to_string_w((bound_buffer as ULen + 20) as *const WideChar, usize::MAX)
+            );
+            assert_eq!(
+                "cccc",
+                input_text_to_string_w((bound_buffer as ULen + 40) as *const WideChar, usize::MAX)
+            );
+            assert_eq!(
+                "dddd",
+                input_text_to_string_w((bound_buffer as ULen + 60) as *const WideChar, usize::MAX)
+            );
+
+            // assert that the indicator has the correct values for column 2.
+            let indicator = s
+                .bound_cols
+                .read()
+                .unwrap()
+                .as_ref()
+                .unwrap()
+                .get(&2)
+                .unwrap()
+                .length_or_indicator;
+            assert_eq!(8, *indicator);
+            assert_eq!(8, *((indicator as ULen + 8) as *mut Len));
+            assert_eq!(8, *((indicator as ULen + 16) as *mut Len));
+            assert_eq!(8, *((indicator as ULen + 24) as *mut Len));
```

Comment:
For some reason, I get `16` when running this locally, but I get `8` when running on evergreen. This also happens with this test: `api::get_info_tests::unit::sql_database_name`; however, I never modified anything that touches that test. I think a `WideChar` can either be 2 or 4 bytes, so that might be what's going on.